src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -106,4 +113,29 @@ public void updateBook(@RequestBody Book book, @PathVariable Long id) {
             .orElseThrow(() -> new EntityNotFoundException(Book.class));
         bookRepository.save(book);
     }
+
+    @GetMapping("/isbn/{isbn}")
+    @ResponseStatus(HttpStatus.OK)
+    public ResponseEntity findByIsbn(@PathVariable String isbn){
+        Optional<Book> book = bookRepository.findByIsbn(isbn);
+        if (book.isPresent())
+            return new ResponseEntity(book.get(), HttpStatus.OK);
+
+        OpenLibraryBookDTO bookDto = openLibrary.tryGetBookByIsbn(isbn);
+        if (bookDto == null)
+            throw new EntityNotFoundException(Book.class);
```

Comment:
Remember that it's better to use the brackets on one line ifs
[Link](https://docs.google.com/document/d/1xBoy41sG9vkrVsOd109OQ61LBQxPzZ2_F3K9e5jfHQY/edit?ts=5e4db74c#heading=h.4bnl8u3wizvm)

Anyway, I think we can make this a little bit different:
- We can set tryGetBookByIsbn as a private method, which can be used by a public one getBookByIsbn, which can return an Optional<Book> (The method signature could be something like this: `public Optional<Book> getBookByIsbn(String isbn)`)
- Having that method I think we can do something like this in the controller:
```
Book book = bookRepository.findByIsbn(isbn)
  .orElse(openLibrary.getBookByIsbn(isbn))
  .orElseThrow(()-> new EntityNotFoundException(Book.class));
```

This way we can have all that logic inside the service, and the controller will act more as a wire between the request and the service :)

Reply:
Wonderful, done! thanks!